lib/compat/wordpress-6.5/blocks.php

```diff
@@ -217,3 +217,190 @@ function gutenberg_process_block_bindings( $block_content, $parsed_block, $block
+	 * @param array $metadata Metadata provided for registering a block type.
+	 * @return array The block type settings
+	 */
+	function gutenberg_filter_block_type_metadata_settings_register_view_styles( $settings, $metadata ) {
+		if ( empty( $metadata['viewStyle'] ) ) {
+			return $settings;
+		}
+		$styles           = $metadata['viewStyle'];
+		$processed_styles = array();
+		if ( is_array( $styles ) ) {
+			for ( $index = 0; $index < count( $styles ); $index++ ) {
+				$result = gutenberg_register_block_view_style_handle(
+					$metadata,
+					$index
+				);
+				if ( $result ) {
+					$processed_styles[] = $result;
+				}
+			}
+		} else {
+			$result = gutenberg_register_block_view_style_handle(
+				$metadata
+			);
+			if ( $result ) {
+				$processed_styles[] = $result;
+			}
+		}
+
+		if ( ! empty( $processed_styles ) ) {
+			$settings['view_style_handles'] = $processed_styles;
+		}
+		return $settings;
+	}
+
+	add_filter( 'block_type_metadata_settings', 'gutenberg_filter_block_type_metadata_settings_register_view_styles', 10, 2 );
+
+	/**
+	 * Enqueue view styles associated with the block.
+	 *
+	 * @param string   $block_content The block content.
+	 * @param array    $block         The full block, including name and attributes.
+	 * @param WP_Block $instance      The block instance.
+	 */
+	function gutenberg_filter_render_block_enqueue_view_styles( $block_content, $parsed_block, $block_instance ) {
+		$block_type = $block_instance->block_type;
+
+		if ( ! empty( $block_type->view_style_handles ) ) {
+			foreach ( $block_type->view_style_handles as $view_style_handle ) {
+				wp_enqueue_style( $view_style_handle );
+			}
+		}
+
+		return $block_content;
+	}
+
+	add_filter( 'render_block', 'gutenberg_filter_render_block_enqueue_view_styles', 10, 3 );
+
+	function gutenberg_register_view_style_rest_field() {
+		register_rest_field(
+			'block-type',
+			'view_style_handles',
+			array(
+				'get_callback' => function ( $item ) {
+					$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $item['name'] );
+					if ( isset( $block_type->view_style_handles ) ) {
+						return $block_type->view_style_handles;
+					}
+					return array();
+				},
+			)
+		);
+	}
+
+	add_action( 'rest_api_init', 'gutenberg_register_view_style_rest_field' );
+
+	/**
+	 * A compat version of `register_block_style_handle()` to use the viewStyle handle.
+	 * Required for core versions < 6.5, since a custom version of generate_block_asset_handle has to be used
+	 * that supports the viewStyle property and correctly creates the handle
+	 *
+	* @param array  $metadata   Block metadata.
+	* @param int    $index      Optional. Index of the style to register when multiple items passed.
+	*                           Default 0.
+	* @return string|false Style handle provided directly or created through
+	*                      style's registration, or false on failure.
+	*/
```

Comment:
Nit, spacing misaligned:
```suggestion
	 * @param array  $metadata   Block metadata.
	 * @param int    $index      Optional. Index of the style to register when multiple items passed.
	 *                           Default 0.
	 * @return string|false Style handle provided directly or created through
	 *                      style's registration, or false on failure.
	 */
```